### ISSUE TITLE
Thick black window border around popup windows and right click menus.

### ISSUE THREAD

**Issue by nikolaof** [Xpra-org/xpra#4380]
**Describe the bug**
The following behavior suddenly appeared at two apps that I'm running through xpra (vscode and planify).
What  happens is when I right click somewhere  or a pop up window opens there is a thick black border on it (see attached image).

**System Information (please complete the following information):**
 - Server OS: Arch linux
 - Client OS: Arch linux
 - Xpra Server Version: 6.1.3
 - Xpra Client Version: 6.1.3
 
**Additional context**

![2024-10-10-174002_377x418_scrot](https://github.com/user-attachments/assets/b96e9c54-e0fc-4d60-a7d8-b2b62701271e)

**Comment by totaam** [comment]
What app can I test with?
This looks like an xshape issue (#501).

Is this a wayland DE?
I don't think we support the shape on other platforms than X11:
https://github.com/Xpra-org/xpra/blob/57777eda2dc6246d06ed42dd1816627af1acfdf4/xpra/client/gtk3/window_base.py#L1256-L1259

**Comment by nikolaof** [comment]
I'm using X11 and so far I saw this on `visual studio code insiders` and `planify` applications

**Comment by totaam** [comment]
Any real open-source applications?

**Comment by nikolaof** [comment]
Isn't [planify](https://github.com/alainm23/planify)  real open source application? It comes with a GPL-3.0 license.

I tested it with obsidian as well and got the same behavior to some right clicks.
Zotero on the other hand does not make it.

**Comment by nikolaof** [comment]
Can you tell me an app that you can use?

**Comment by totaam** [comment]
Any application in the Fedora repositories -  there are thousands of them.

**Comment by nikolaof** [comment]
So it is happening on Visual Studio code. Can you test it on this app?

**Comment by totaam** [comment]
No `vscode` is not the Fedora repos.

> Is this a wayland DE? Does it support compositing?

Maybe your window manager handle transparent windows?
Please see: https://github.com/Xpra-org/xpra/wiki/Reporting-Bugs

**Comment by nikolaof** [comment]
> No `vscode` is not the Fedora repos.
> 
> > Is this a wayland DE? Does it support compositing?
> 
> Maybe your window manager handle transparent windows? Please see: https://github.com/Xpra-org/xpra/wiki/Reporting-Bugs

I'm using i3 on X11. I installed picom and the difference is that it shows this thick area but now kind of transparent.

**Comment by totaam** [comment]
> but now kind of transparent.

Can you include a screenshot with and without picom?

**Comment by nikolaof** [comment]
Without picom is the image in the initial post and here is the one with picom

![2024-11-21-134638_346x427_scrot](https://github.com/user-attachments/assets/46845e40-3a91-4525-bdb7-fd7a23538b9e)

**Comment by totaam** [comment]
This popup menu looks an awful lot like the one the shows up when you click on `gedit`'s pancake button.
Can you run `gedit` to see if you get the same problem?

**Comment by nikolaof** [comment]
Nope. this is not the case with the gedit. I tried it and it works fine with and without picom.
I start think that the issue is only related with apps that are written using electron framework

**Comment by totaam** [comment]
> the issue is only related with apps that are written using electron framework

That's quite possible, likely even: https://github.com/Xpra-org/xpra/issues/4365#issuecomment-2370004460
